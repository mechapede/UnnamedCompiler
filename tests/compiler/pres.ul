void operate_all ( int x, char y, string j, float z ) {
    int xxx;
    char zzz;
    float fff;
    fff = 10.0 + z * 3.0;
    if ( z * 3.0 + 11.0 < fff ) {
        println "Precedence of float failed, unless overflow!";
    }
    zzz = 'a' + y;
    xxx = x + 101 * 3 * 2;
}

void bop(){

}

//a normal comment to test this
void main () {
    int x;
    int y;
    x = 1 + 3*4*11 + 7 + 6 * 2;
    y = 3*4*11 + 1 + 7 + 6 * 2;
    if ( x == y ) {

    } else {
        println "Pres problems with PLUS and MINUS";
    }

    if( 1 < 2 == 1 < 2 ) {

    } else {
        print "Something wrong with LESSTHAN and EQUALEQUAL";
    }

    if ( 2 * 4 + 3 == 3 + 2 * 4 ) {

    } else {
        print "Issues with PLUS, MULTI and EQUALEQUAL";
    }

    if ( 2 * 4 - 3 == 0 - 3 + 2 * 4 ) {

    } else {
        print "Issues with PLUSMINUS, MULTIPLY and EQUALEQUAL{";
    }
    
    if ( 1.0 == 1.0 ){
      println "float comparisons works";
    } 
    
    if ( 2.0 == 1.0 ){
      println "float comparisons wacky";
    } 
    
    bop();
    
    print "all done ";
    println true;

}
