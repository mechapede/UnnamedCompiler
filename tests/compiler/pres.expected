float comparisons works
all done true
